### PR TITLE
implement royalty enforcement optional

### PR DESCRIPTION
**Pull request by kneelsdev**

### CODE REVIEW

File: contracts/Archetype.sol [hunk 1]
```diff
@@ -433,7 +438,7 @@ contract Archetype is ERC721A__Initializable, ERC721AUpgradeable, ERC721A__Ownab
   }
```
Comment (coindegen) [inline contracts/Archetype.sol:438]: can we rename the `reveal()` function to something more accurate like `toggleRevealed()`? Currently misleading

Comment (h8ku) [reply contracts/Archetype.sol:438]: not sure if we even need it anymore, it is part of an old design pattern we don't use anymore

**Comment by kneelsdev** [comment]
Refactor gas improvements:
before: (1000 addresses) = 52019374
after: (1000 addresses) = 48081576

File: contracts/Archetype.sol [hunk 2]
```diff
@@ -185,17 +188,29 @@ contract Archetype is ERC721A__Initializable, ERC721AUpgradeable, ERC721A__Ownab
...
     bytes calldata signature
   ) public payable {
-    if(quantity.length != to.length) {
+    if(quantityList.length != toList.length) {
       revert InvalidConfig();
     }
-    for (uint256 i=0; i< to.length; i++){
-      mintTo(auth, quantity[i], to[i], affiliate, signature);
+    uint256 quantity = 0;
+    for (uint256 i=0; i< quantityList.length; i++){
```
Comment (coindegen) [inline contracts/Archetype.sol:200]: How costly is this loop is in gas terms? Just out of curiosity

Comment (kneelsdev) [reply contracts/Archetype.sol:200]: It costs 123,128 gas units to run that for 1000 addresses/quantities
(123 gas units per loop)

Which is fairly cheap.

its when you loop through storage writes/reads that things get expensive.

**Pull request merged**